source/Xunit.Gherkin.Quick/CoreModel/Feature/Discovery/FeatureDiscoveryModel.cs

```diff
@@ -11,18 +13,31 @@ public FeatureDiscoveryModel(IFeatureFileRepository featureFileRepository)
             _featureFileRepository = featureFileRepository ?? throw new ArgumentNullException(nameof(featureFileRepository));
         }
 
-        public global::Gherkin.Ast.Feature Discover(Type featureClassType)
+        public IEnumerable<Tuple<string,global::Gherkin.Ast.Feature>> Discover(Type featureClassType)
         {
             if (featureClassType == null)
                 throw new ArgumentNullException(nameof(featureClassType));
 
-            var fileName = FeatureClassInfo.FromFeatureClassType(featureClassType).FeatureFilePath;
-            var featureFile = _featureFileRepository.GetByFilePath(fileName);
-            if (featureFile == null)
-                throw new System.IO.FileNotFoundException("Feature file not found.", fileName);
+            var featureClassInfo = FeatureClassInfo.FromFeatureClassType(featureClassType);
+            var fileNameSearchPattern = featureClassInfo.FileNameSearchPattern;

```

Comment:
I don't follow. You are not using the fileNameSearchPattern in this method, but you should.

What you should have in this method is this:
1. Find the feature file using featureClassInfo.FeatureFilePath (this already worked, so it's already there before your changes).
2. Find feature files using featureClassInfo.FileNameSearchPattern.
3. Union 1 and 2 from above.
4. Create Gherkin.Ast.Feature for all those files. and return that.

What you have works perhaps because you passed the file name search pattern into the repository, which is a bad idea because you have created tight coupling between the discoverers and this class. The caller should not help the called method do its job, that's tight coupling. All the logic must be here, not outside.